compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala

```diff
@@ -313,6 +314,10 @@ class ClassfileParser(
       val isVarargs = denot.is(Flags.Method) && (jflags & JAVA_ACC_VARARGS) != 0
       denot.info = sigToType(sig, isVarargs = isVarargs)
       if (isConstructor) normalizeConstructorParams()
+      if isNative then
```

Comment:
To match the behavior of Scala 2 it looks like we should do the same for transient and volatile: https://github.com/scala/scala/blob/f8a5497df5303cf5ba4a7af545aed326556c604f/src/reflect/scala/reflect/internal/JavaAccFlags.scala#L75-L77
We already do it when parsing .java source file so that can stay as is:
https://github.com/lampepfl/dotty/blob/33818506801c80c8c73649fdaab3782c052580c6/compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala#L475-L480